File: app/src/main/java/com/nytimes/android/external/register/model/ConfigSku.java
package com.nytimes.android.external.register.model;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

@Value.Immutable
@Gson.TypeAdapters
public abstract class ConfigSku {

    public abstract String type();
    public abstract String price();
    public abstract String title();
    public abstract String description();

    @Value.Default
    @SuppressWarnings("PMD.EmptyMethodInAbstractClassShouldBeAbstract")
    public int introductoryPriceCycles() {
        return 0;
    }

    @Nullable
    public abstract String freeTrialPeriod();
    @SerializedName("package")
    public abstract String packageName();
    @Nullable
    public abstract String introductoryPrice();
    @Nullable
    public abstract String introductoryPricePeriod();
    @Nullable
    public abstract String subscriptionPeriod();
}

File: app/src/main/java/com/nytimes/android/external/register/model/ConfigSku.kt
package com.nytimes.android.external.register.model

import com.google.gson.annotations.SerializedName

data class ConfigSku(
        val type: String,
        val price: String,
        val title: String,
        val description: String,
        @SerializedName("package")
        val packageName: String,
        val introductoryPriceCycles: Int = 0,
        val freeTrialPeriod: String? = null,
        val introductoryPrice: String? = null,
        val introductoryPricePeriod: String? = null,
        val subscriptionPeriod: String? = null
)